projects/ad7124-4sdz/src/ad7124-4sdz.c

```diff
@@ -0,0 +1,64 @@
+#include <stdio.h>
+#include <inttypes.h>
+#include "ad7124.h"
+#include "ad7124_regs.h"
+#include "spi_extra.h"
+
+/******************************************************************************/
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
+
+int main(void)
+{
+	struct ad7124_dev *dev;		/* A driver handle to pass around */
+	enum ad7124_registers regNr;	/* Variable to iterate through registers */
+	long timeout = 1000;		/* Number of tries before a function times out */
+	long ret = 0;			/* Return value */
+	long sample;			/* Stores raw value read from the ADC */
+
+	static struct xil_spi_init_param xil_spi_init_params = {
+		.type = SPI_PS,
+	};
+
+	struct spi_init_param spi_init = {
+		.platform_ops = &xil_platform_ops,
+		.chip_select = 0,
+		.max_speed_hz = 1000000,
+		.mode = SPI_MODE_3,
+		.extra = &xil_spi_init_params,
+	};
+
+	struct ad7124_init_param ad7124_init_param = {
+		.spi_init = &spi_init,
+		.regs = ad7124_regs,
+		.spi_rdy_poll_cnt = 25000,
+	};
+
+	/* Initialize AD7124 device. */
+	ret = ad7124_setup(&dev, &ad7124_init_param);
+	if (ret < 0)
```

Comment:
if the return value for the function is either 0 or a negative value, we can use `ret != SUCCESS`

Reply:
Check through the entire code.